### ISSUE TITLE
`context: scope` does not recognise if context was declared in function

### ISSUE THREAD

**Issue by makupi** [go-simpler/sloglint#91]
```go
package main

import (
	"context"
	"log/slog"
)

func main() {
	foo()
}

func foo() {
	ctx := context.Background()
	slog.Error("Failed to do X")
	slog.ErrorContext(ctx, "Failed to do X with ctx")
}
```
With a config like
```yml
version: "2"

linters:
  enable:
    - sloglint
  settings:
    sloglint:
      context: "scope"
```
Produces 0 issues.
```
$ golangci-lint run
0 issues.
```

If I move the context to a function parameter:
```go
package main

import (
	"context"
	"log/slog"
)

func main() {
	foo(context.Background())
}

func foo(ctx context.Context) {
	slog.Error("Failed to do X")
	slog.ErrorContext(ctx, "Failed to do X with ctx")
}
```
```
$ golangci-lint run
main.go:13:2: ErrorContext should be used instead (sloglint)
        slog.Error("Failed to do X")
        ^
1 issues:
* sloglint: 1
```

Versions:
- golangci-lint: 2.2.0
- go: 1.23.8

**Comment by makupi** [comment]
Ah right. My use-case is opentelemetry, I'm creating the span in the function scope and I would like to have the span id in logs from context. 
```go
ctx, span := tracer.Start(context.Background(), "span")
defer span.End()

doSomething(ctx)
slog.ErrorContext(ctx, "msg") // i would like to have this call reported, as the context is still valid
```
But this is probably too specific so I understand, might have to stick with `context: all`

**Comment by tmzane** [comment]
I usually start a new span from an existing context, like this:

```go
ctx, span := tracer.Start(ctx, "span")
defer span.End()
```

In case there is already a span in the context, tracer.Start simply returns it, otherwise it starts a new span. The context is passed as a function param, so `context: scope` works well.
